src/Algebra/Consequences/Setoid.agda

```diff
@@ -48,6 +48,24 @@ module _ {_•_ : Op₂ A} (comm : Commutative _•_) where
     x • z ≈⟨ comm x z ⟩
     z • x ∎)
 
+  comm+cancelˡ-nonZero⇒cancelʳ-nonZero :
+    (0# : A) → AlmostLeftCancellative 0# _•_ → AlmostRightCancellative 0# _•_
+  comm+cancelˡ-nonZero⇒cancelʳ-nonZero 0# cancelˡ-nonZero {x} y z x≉0 yx≈zx =
+    cancelˡ-nonZero y z x≉0 (begin
```

Comment:
You can use `$` instead of parentheses around the `begin (...) ∎` block.

Reply:
With `$`, it is not parsed. 
In the cases when  `begin` directly follows after "=",  `$` is parsed.

Reply:
Really? It's used in many places such as
* Data.Nat.Properties
* Data.Nat.Divisibility
* Data.Nat.Binary.Subtraction
* Data.Digit
etc., etc.

Reply:
Well, try to find anywhere  ``= <foo> $ begin ``, 
where <foo> is not empty.

Reply:
I gave a list of modules where you had such examples.
For instance http://agda.github.io/agda-stdlib/Data.Nat.Divisibility.html#10002

```agda
nonZeroDivisor-lemma m zero r r≢zero (divides (suc q) eq) =
  m+1+n≰m m $ begin
    m + suc (q * suc m) ≡⟨ +-suc m (q * suc m) ⟩
    suc (m + q * suc m) ≡⟨ sym eq ⟩
    1 * toℕ r           ≡⟨ *-identityˡ (toℕ r) ⟩
    toℕ r               ≤⟨ FP.toℕ≤pred[n] r ⟩
    m                   ∎
```

Reply:
So, I do not know, why this $ is parsed in one examples and is not in others.

Reply:
Did you forget to add `open import Function.Base using (_$_)` in the import list?

Reply:
Yes, I did. I am sorry.
Now I set this `$` to all these places, except the ones after comma
(because ``foo , $ begin ...`` does not look nice, and also I am not sure that this will be parsed,
also the parentheses often allow to skip the import of  the module Function).
?